base_station/gui/src/components/ToggleButton.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+    <label :for="id + '_button'" :class="{'active': currentState}" class="toggle__button">
+        <span v-if="currentState" class="toggle__label">{{ labelEnableText }}</span>
+        <span v-if="! currentState" class="toggle__label">{{ labelDisableText }}</span>
+
+        <input type="checkbox" :id="id + '_button'" v-model="checkedValue">
+        <span class="toggle__switch"></span>
+    </label>
+</template>
+
+<script>
+export default {
+    data() {
+        return {
+            currentState: this.currentState
+        }
+    },
+
+    props: {
+        labelEnableText: {
+            type: String,
+            required: true
+        },
+        labelDisableText: {
+            type: String,
+            required: true
+        },
+        id: {
+            type: String,
+            default: 'primary'
+        }, 
+        currentState: {
+            type: Boolean,
+            required: true
+        }
+    },
+
+    computed: {
+        checkedValue: {
+            get() {
+                return this.currentState;
+            },
+            set(newValue) {
+                this.currentState = newValue;
```

Comment:
Not sure this is necessary. It's throwing a bunch of console errors because you're not supposed to set the value of a prop. I think it should be fine to just emit the change, and the parent of this component will update the currentState, which should carry down to this component as a prop